src/middlewared/middlewared/plugins/smb_/registry_share.py

```diff
@@ -372,6 +373,9 @@ async def share_to_smbconf(self, conf_in, globalconf=None):
         else:
             conf["nt acl support"] = "no"
 
+        if data["cluster_volname"]:
```

Comment:
I've done an initial pass at reviewing this and syntactically, everything looks fine. I think this is where we should `cache.put SMB_HA_MODE` logic so the cache is updated making `smb_get_ha_mode` return clustered.

I'll review a lot more in depth tomorrow.